tests/test_cnes_sonar_scanner.py

```diff
@@ -0,0 +1,517 @@
+                        auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD)).json()
+        # Hint: if this test fails, the project is not on the server
+        assert output['components'][0]['key'] == project_key
+        # Get the number of issues of the project
+        output = requests.get(f"{cls.SONARQUBE_LOCAL_URL}/api/issues/search?componentKeys={project_key}",
+                        auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD)).json()['issues']
+        issues = [ issue for issue in output if issue['status'] in ('OPEN', 'TO_REVIEW') ]
+        # Hint: if this test fails, there should be {nb_issues issues} on the {language_name} dummy project with the Sonar way QP but {len(issues)} were found
+        assert len(issues) == nb_issues
+        print("Analysis with Sonar way QP ran as expected.")
+        # If the language has a specific CNES Quality Profile, it must also be tested
+        if cnes_qp:
+            # Switch to CNES QP
+            requests.post(f"{cls.SONARQUBE_LOCAL_URL}/api/qualityprofiles/add_project",
+                auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD),
+                data={
+                    "language": language_key,
+                    "project": project_key,
+                    "qualityProfile": cnes_qp
+                })
+            # Rerun the analysis
+            docker_client.containers.run(cls._SONAR_SCANNER_IMAGE, f"-Dsonar.projectBaseDir=/usr/src/tests/{folder}",
+                auto_remove=True,
+                environment={"SONAR_HOST_URL": cls.SONARQUBE_URL},
+                network=cls.SONARQUBE_NETWORK,
+                user=f"{os.getuid()}:{os.getgid()}",
+                volumes={
+                    f"{cls._PROJECT_ROOT_DIR}": {'bind': '/usr/src', 'mode': 'rw'},
+                    f"{cls._PROJECT_ROOT_DIR}/.sonarcache": {'bind': '/opt/sonar-scanner/.sonar/cache', 'mode': 'rw'}
+                })
+            # Wait for SonarQube to process the results
+            time.sleep(8)
+            # Switch back to the Sonar way QP (in case the test needs to be rerun)
+            requests.post(f"{cls.SONARQUBE_LOCAL_URL}/api/qualityprofiles/add_project",
+                auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD),
+                data={
+                    "language": language_key,
+                    "project": project_key,
+                    "qualityProfile": "Sonar way"
+                })
+            # Get the new number of issues
+            output = requests.get(f"{cls.SONARQUBE_LOCAL_URL}/api/issues/search?componentKeys={project_key}",
+                auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD)).json()['issues']
+            issues = [ issue for issue in output if issue['status'] in ('OPEN', 'TO_REVIEW') ]
+            # Hint: if this test fails, there should be {nb_issues_cnes_qp} issues on the {language_name} dummy project with the {cnes_qp} QP but {len(issues)} were found
+            assert len(issues) == nb_issues_cnes_qp
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonar-scanner/pullRequest?prid=6252273)